apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1/MetadataClient.cs

```diff
@@ -291,7 +289,7 @@ public virtual Task<bool> IsServerAvailableAsync(CancellationToken cancellationT
         /// <seealso cref="GetInstanceMetadataAsync"/>
         /// <seealso cref="WaitForChange"/>
         /// <seealso cref="WaitForChangeAsync"/>
-        public virtual event EventHandler InstanceMetadataChanged
+        public virtual event EventHandler<Instance> InstanceMetadataChanged
```

Comment:
It's slightly unusual to use the entity type itself as the `TEventArgs` in an `EventHandler<T>`, although it certainly works. An alternative would be to have a `MetadataChangedEventArgs<T>`, and use `EventHandler<MetadataChangedEventArgs<Project>>` etc. That would allow future expansion for (say) timestamp of the change being noticed or similar.

Definitely not a blocking comment, but worth considering.